lib/plugins/aws/deploy/lib/mergeIamTemplates.js

```diff
@@ -11,6 +11,47 @@ module.exports = {
   },
 
   merge() {
+    // resolve early if no functions are provided
+    if (!this.serverless.service.getAllFunctions().length) {
+      return BbPromise.resolve();
+    }
+
+    // resolve early if provider level role is provided
+    if ('role' in this.serverless.service.provider) {
+      return BbPromise.resolve();
+    }
+
+    // resolve early if all functions contain a custom role
+    const customRolesProvided = [];
+    this.serverless.service.getAllFunctions().forEach((functionName) => {
+      const functionObject = this.serverless.service.getFunction(functionName);
+      customRolesProvided.push('role' in functionObject);
+    });
+    if (_.isEqual(_.uniq(customRolesProvided), [true])) {
+      return BbPromise.resolve();
+    }
+
+    // merge in the iamRoleLambdaTemplate
+    const iamRoleLambdaExecutionTemplate = this.serverless.utils.readFileSync(
+      path.join(this.serverless.config.serverlessPath,
+        'plugins',
+        'aws',
+        'deploy',
+        'lib',
+        'iam-role-lambda-execution-template.json')
+    );
+    iamRoleLambdaExecutionTemplate.Properties.Path = this.provider.naming.getRolePath();
+    iamRoleLambdaExecutionTemplate.Properties.RoleName = this.provider.naming.getRoleName();
+    iamRoleLambdaExecutionTemplate.Properties.Policies[0]
+      .PolicyName = this.provider.naming.getPolicyName();
+
+    _.merge(
+      this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
+      {
+        [this.provider.naming.getRoleLogicalId()]: iamRoleLambdaExecutionTemplate,
+      }
+    );
+
     this.serverless.service.getAllFunctions().forEach((functionName) => {
```

Comment:
I'm not sure if I completely follow because of all the prior checks:
```js
// resolve early if no functions are provided
// resolve early if provider level role is provided
// resolve early if all functions contain a custom role
```

If we're talking purely about Log groups and assuming the user is providing his own IAM statement to put logs the following code block:
```js
this.serverless.service.getAllFunctions().forEach((functionName) => {
      const functionObject = this.serverless.service.getFunction(functionName);
      const logGroupLogicalId = this.provider.naming
        .getLogGroupLogicalId(functionName);
      const newLogGroup = {
        [logGroupLogicalId]: {
          Type: 'AWS::Logs::LogGroup',
          Properties: {
            LogGroupName: this.provider.naming.getLogGroupName(functionObject.name),
          },
        },
      };
      _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
        newLogGroup);
});
```
might come between:
```js
// resolve early if no functions are provided
// HEREEEEEE
// resolve early if provider level role is provided
```